java/vector/src/main/java/org/apache/arrow/vector/complex/BaseListVector.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.complex;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.BaseValueVector;
+import org.apache.arrow.vector.FieldVector;
+
+/**
+ * Abstraction for all list type vectors.
+ */
+public abstract class BaseListVector extends BaseValueVector implements PromotableVector {
+
+  protected BaseListVector(BufferAllocator allocator) {
+    super(allocator);
+  }
+
+  /**
```

Comment:
It seems like all we really need here is to be able to clone the existing vector and get/set the data vector.
   It cloning already available through TransferPair?  the getStartEndIndex are probably useful as well but I don't like the setOffsetBufferValueIfNeeded.

Reply:
Awesome, use clone ListVector and reset data vector logic instead.

Reply:
Actually, looking more closely I think everything that is needed other then getElementStartIndex and getElementEndIndex is available through FieldVector (see getBuffers and getChildrenVectors) so I don't think you need these fields.

Reply:
Sure, we can make BaseListVector extends FieldVector and keep getElementStartIndex and getElementEndIndex.
We could get data vector via FieldVector#getChildrenVectors, and seems reset data vector is not available, should we remain setDataVector here?